### PR TITLE
EVA - 3551 - Upload json to submit

### PR DESCRIPTION
**Pull request by tcezard**

### CODE REVIEW

**Review by nitin-ebi — APPROVED** [review]

File: eva_sub_cli/main.py [hunk 1]
```diff
@@ -55,15 +55,19 @@ def orchestrate_process(submission_dir, vcf_files_mapping, vcf_files, assembly_f
...
-                                 submission_config=sub_config) as validator:
-                validator.validate_and_report()
+            validator = NativeValidator(vcf_files_mapping, submission_dir, metadata_json, metadata_xlsx,
+                                 submission_config=sub_config)
+        with validator:
+            validator.validate_and_report()
+            if not metadata_json:
+                metadata_json = os.path.join(validator.output_dir, 'metadata.json')
+                sub_config.set('metadata_json', value=metadata_json)
+                sub_config.set('vcf_files', value=vcf_files)
```
Comment (apriltuesday) [inline eva_sub_cli/main.py:69]: Should these be outside the `if`?

File: tests/test_main.py [hunk 2]
```diff
@@ -38,9 +38,8 @@ def test_orchestrate_validate(self):
                 self.mapping_file, self.test_sub_dir, self.metadata_json, self.metadata_xlsx,
                 submission_config=m_config.return_value
             )
-
-            with m_docker_validator() as validator:
-                validator.validate_and_report.assert_called_once_with()
+            print(m_docker_validator.mock_calls)
```
Comment (apriltuesday) [inline tests/test_main.py:41]:
```suggestion
```

File: eva_sub_cli/main.py (hunk 1 above)
Comment (tcezard) [reply eva_sub_cli/main.py]: Yes they should.

**Review by apriltuesday — APPROVED** [review]
The test failure is odd, otherwise this looks fine to me.

**Comment by tcezard** [comment]
Test failure was due to the docker image v0.0.1.dev6 in dockerhub not containing the md5sum generation added in https://github.com/EBIvariation/eva-sub-cli/pull/30.
I overwrote the image and it now passes.

**Pull request merged**